### PR TITLE
[GTKUI] Added detection of torrent URL on GTK UI focus 

### PR DESCRIPTION
**Pull request by matiasw**
There is a feature in Transmission Remote GUI that pops up the "Add torrent" dialog automatically if it detects that there is a torrent URL on the clipboard when the program gets focus. This is a preliminary implementation of that feature in Deluge.

At the moment, it works, but it does not have the intelligence to check if the torrent on the clipboard is already being downloaded. On IRC, I was adviced to "grab the .torrent and check to see if the hash already exists", but this seems like a rather heavy operation to perform each time the program gets focus with a torrent URL on the clipboard.

### CODE REVIEW

**Review by DjLegolas — CHANGES_REQUESTED** [review]
Please fix the code formatting errors (from `lint`):
1. fix `import`s order
2. use only single-quote

Best way is to run the `tox` tests yourself.

In addition, please change the commit message as it is too long for one line.
Something like:
```
[GTKUI] Added detection of torrent URL on GTK UI focus

In case deluge GTK gets focus with a new torrent URL on the clipboard, the "Add Torrent from URL" dialog will pop up automatically
```

**Comment by matiasw** [comment]
@DjLegolas Thanks for the feedback. I changed quotes into single quotes and changed the import order; not sure if it's still the way you'd like?

File: deluge/ui/gtk3/mainwindow.py [hunk 1]
```diff
@@ -12,9 +12,10 @@
 import logging
 import os.path
 from hashlib import sha1 as sha
+from urllib.parse import urlparse
 
 import gi
-from gi.repository import Gtk
+from gi.repository import Gtk, Gdk
```
Comment (DjLegolas) [inline deluge/ui/gtk3/mainwindow.py:18]: Just switch the order like this:
```python
from gi.repository import Gdk, Gtk
```

**Review by DjLegolas — CHANGES_REQUESTED** [review]

File: deluge/ui/gtk3/mainwindow.py (hunk 1 above)
Comment (matiasw) [reply deluge/ui/gtk3/mainwindow.py]: Ok, done.

**Comment by Erarshad** [comment]
https://github.com/CCExtractor/Deluge-mobile-remote-client
now deluge mobile client is also available for android and ios. it is open source fork repository give star and feel free to send PR

**Review by cas-- — CHANGES_REQUESTED** [review]

**Comment by matiasw** [comment]
Ok, so I changed the code to use these premade functions. It is a bit neater, now.

**Review by cas-- — CHANGES_REQUESTED** [review]
Looking much better, can we do some final linting cleanup. Use `isort` to auto-fix the linting issues. 

I was just testing it out and wonder if this should be a user preference since without any way for the user to disable it might be annoying. 

You can search for the `gtkui_config['show_piecesbar']` preference to use as a guide to implementing it.

image

**Comment by matiasw** [comment]
I have made the requested modifications and added the Preferences setting for this feature.

**Comment by cas--** [comment]
Excellent, thanks! Just need to fix the failing linting issues

image

If you install [pre-commit](pre-commit.com) then your commits are linted automatically

**Comment by matiasw** [comment]
Uh, so, I did install pre-commit, and on my machine, with `pre-commit run --all-files`, I get all PASSED, but the CI test on Github still fails. If I knew why, I could fix it?

**Comment by cas--** [comment]
That is strange, I see this: 

image

Don't worry I'll can resolve it when merging it :+1:

**Comment by cas--** [comment]
Thanks @matiasw for this useful feature!

**Pull request closed without merging**